core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -519,44 +523,43 @@ private[log] class Cleaner(val id: Int,
   private def compressMessages(buffer: ByteBuffer,
                                compressionCodec: CompressionCodec,
                                messageFormatVersion: Byte,
-                               messageAndOffsets: Seq[MessageAndOffset]) {
+                               messageAndOffsets: Seq[MessageAndOffset]): Int = {
     require(compressionCodec != NoCompressionCodec, s"compressionCodec must not be $NoCompressionCodec")
-    if (messageAndOffsets.nonEmpty) {
```

Comment:
Why did you remove this `if`? The methods below assume that the `messageAndOffsets` are not empty. If you are sure that we never call it with an empty Seq, then we should probably add a `require` to make that explicit.